src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs

```diff
@@ -0,0 +1,214 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using System.Windows.Forms.Internal;
+using Xunit;
+using static Interop;
+
+namespace System.Windows.Forms.Tests.Text
+{
+    public class FontMetrics
+    {
+        [Theory()]
+        [InlineData("Arial", 9.0f, 15)]
+        [InlineData("Arial", 12.0f, 18)]
+        [InlineData("Microsoft Sans Serif", 16.0f, 26)]
+        [InlineData("Times New Roman", 11.0f, 17)]
+        [InlineData("MS Gothic", 10.0f, 14)]
+        public void Font_GetHeight(string family, float size, int height)
+        {
+            using Font font = new Font(family, size);
+            if (font.Name != family)
+            {
+                // Not installed on this machine
+                return;
+            }
+
+            using WindowsFont windowsFont = WindowsFont.FromFont(font, Gdi32.QUALITY.CLEARTYPE);
+            Assert.Equal(height, windowsFont.Height);
+        }
+
+        [Theory()]
+        [InlineData("Arial", 9.0f, 1 /* DEFAULT_CHARSET */)]
+        [InlineData("Arial", 12.0f, 1)]
+        [InlineData("Microsoft Sans Serif", 16.0f, 1)]
+        [InlineData("Times New Roman", 11.0f, 1)]
+        [InlineData("MS Gothic", 11.0f, 1)]
+        public void Font_GetCharSet(string family, float size, byte charset)
+        {
+            using Font font = new Font(family, size);
+            if (font.Name != family)
+            {
+                // Not installed on this machine
+                return;
+            }
+
+            using WindowsFont windowsFont = WindowsFont.FromFont(font, Gdi32.QUALITY.CLEARTYPE);
+            Assert.Equal(charset, windowsFont.CharSet);
+        }
+
+        [Theory()]
+        [InlineData("Arial", 9.0f, 2.5f)]
+        [InlineData("Arial", 12.0f, 3.0f)]
+        [InlineData("Microsoft Sans Serif", 16.0f, 4.3333335f)]
+        [InlineData("Times New Roman", 11.0f, 2.8333333f)]
+        [InlineData("MS Gothic", 10.0f, 2.3333333f)]
+        public void Font_GetOverhangPadding(string family, float size, float expected)
+        {
+            using Font font = new Font(family, size);
+            if (font.Name != family)
+            {
+                // Not installed on this machine
+                return;
+            }
+
+            using WindowsFont windowsFont = WindowsFont.FromFont(font, Gdi32.QUALITY.CLEARTYPE);
+            WindowsGraphics graphics = WindowsGraphicsCacheManager.GetCurrentMeasurementGraphics();
+            Assert.Equal(expected, graphics.GetOverhangPadding(windowsFont));
+        }
+
+        [Theory(),
+            InlineData("Arial", 9.0f, 3, 4),
+            InlineData("Arial", 12.0f, 3, 5),
+            InlineData("Microsoft Sans Serif", 16.0f, 5, 7),
+            InlineData("Times New Roman", 11.0f, 3, 5),
+            InlineData("MS Gothic", 10.0f, 3, 4),
+            ]
```

Comment:
We don't combine the attributes
```suggestion
        [Theory]
        [InlineData("Arial", 9.0f, 3, 4)]
        [InlineData("Arial", 12.0f, 3, 5)]
        [InlineData("Microsoft Sans Serif", 16.0f, 5, 7)]
        [InlineData("Times New Roman", 11.0f, 3, 5)]
        [InlineData("MS Gothic", 10.0f, 3, 4)]
```